modules/nextflow/src/main/groovy/nextflow/scm/AzureDevOpsRepositoryProvider.groovy

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020, Seqera Labs
+ * Copyright 2013-2019, Centre for Genomic Regulation (CRG)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.scm
+
+import groovy.transform.CompileDynamic
+import groovy.transform.CompileStatic
+import groovy.transform.Memoized
+
+/**
+ * Implements a repository provider for Azure DevOps service
+ *
+ * @author Tobias Neumann <tobias.neumann.at@gmail.com>
+ */
+@CompileStatic
+final class AzureDevOpsRepositoryProvider extends RepositoryProvider {
+
+    private String user
+    private String repo
+
+    AzureDevOpsRepositoryProvider(String project, ProviderConfig config=null) {
+        this.project = project
+        this.user = this.project.tokenize('/').first()
+        this.repo = this.project.tokenize('/').last()
+        this.config = config ?: new ProviderConfig('azurerepos')
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    String getName() { "Azure Repos" }
+
+    /** {@inheritDoc} */
+    @Override
+    String getEndpointUrl() {
+        "${config.endpoint}/${project}/_apis/git/repositories/${repo}"
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    String getContentUrl( String path ) {
+        "${config.endpoint}/${project}/_apis/git/repositories/${repo}/items?download=false&includeContent=true&includeContentMetadata=false&api-version=6.0&\$format=json&path=$path"
+    }
+
+    @Override
+    @CompileDynamic
+    List<BranchInfo> getBranches() {
+        // https://docs.microsoft.com/en-us/rest/api/azure/devops/git/refs/list?view=azure-devops-rest-6.0#refs-heads
+        final url = "${config.endpoint}/${project}/_apis/git/repositories/${repo}/refs?filter=heads&api-version=6.0"
```

Comment:
Giving a quick look, should not be Get Branches ? 

https://docs.microsoft.com/en-us/rest/api/azure/devops/tfvc/branches/get%20branches?view=azure-devops-rest-6.0

Reply:
This is a different versioning section - the one you should be looking for is here:

https://docs.microsoft.com/en-us/rest/api/azure/devops/git/?view=azure-devops-rest-6.0

```
Git API
 
Use these APIs to work with Git repositories in Azure DevOps Services. Repositories contain objects representing core Git concepts:

blobs (files)
trees (folders)
commits
refs (branches, lightweight tags)
and other resources that represent the history of those items or act on their state.
```

Also yields nothing when I query it:

```
{"count":0,"value":[]}
```

Reply:
@pditommaso U agree that using the ```git``` section of the API is the proper way to go and we can finalize the PR?